addons/web/views/webclient_templates.xml

```diff
@@ -598,6 +598,21 @@
         </t>
     </template>
 
+    <template id="web.app_auth" name="Accept app">
+        <t t-call="web.login_layout">
+            <form class="oe_login_form" role="form" method="post" action="/web/oauth">
+                <input type="hidden" name="csrf_token" t-att-value="request.csrf_token()"/>
+                <input type="hidden" name="redirect" t-att-value="redirect"/>
+                <input type="hidden" name="scope" t-att-value="scope"/>
+
+                <t t-esc="scope" /> wants to access Odoo.
```

Comment:
There should probably be an option for a clearer application label / name, and optionally a justification / reminder to use as the token name / label? How does e.g. github handler this sort of things?

Reply:
What about adding a new model with three fields: `scope`, `name`, and even a `description`?

Reply:
As we discussed on Discord, I rather added a `friendlyname` and an `info` parameters that should be encoded in the url like this:
http://localhost:8069/web/login?redirect=%2Fweb%2Fauth%2Fapi%2Frequest_key%3Fscope%3Dmail.client.outlook%26friendlyname%3DOutlook%26info%3DFirefox%20Win32%202020-08-10%26redirect%3Dhttps%253A%252F%252Fgoogle.com

Which redirects to such a page: 
![image](https://user-images.githubusercontent.com/372102/89805684-504a8780-db36-11ea-9b34-8b44cd1e6d56.png)

The token description is then a concatenation of the friendly name and the additional information:
![image](https://user-images.githubusercontent.com/372102/89805852-88ea6100-db36-11ea-8473-4f59e9bafc2b.png)